### ISSUE TITLE
Given tf.train.Example, how to get tfds.features?

### ISSUE THREAD

**Issue by zeyuwang615** [tensorflow/datasets#4695]
**What I need help with / What I was wondering**
Hi, I am trying to load external tfrecord files with TFDS. I have read the official doc [here](https://www.tensorflow.org/datasets/external_tfrecord), and find I need to define the feature structure using `tfds.features`. However, since the tfrecords files are alreay generated, I do not have control the generation pipeline. I do, however, know the `tf.train.Example` structre used in `TFRecordWriter` during generation, shown as follows.
`
from tensorflow.python.training.training import BytesList, Example, Feature, Features, Int64List
dict(Example=Features({
'image': Feature(bytes_list=BytesList(value=[img_str])), # img_str is jpg encoded image raw bytes
'caption': Feature(bytes_list=BytesList(value=[caption])), # caption is a string
'height': Feature(bytes_list=Int64List(value=[caption])), # caption is a string
'width': Feature(bytes_list=Int64List(value=[caption])), # caption is a string
})
`
The doc only describes how to translate [tfds.features](https://www.tensorflow.org/datasets/api_docs/python/tfds/features) into the human readable structure of the [tf.train.Example](https://www.tensorflow.org/api_docs/python/tf/train/Example). But nowhere does it mention how to translate a tf.train.Example into tfds.features, which is needed to automatically add the proper metadata fileswith [tfds.folder_dataset.write_metadata](https://www.tensorflow.org/datasets/api_docs/python/tfds/folder_dataset/write_metadata).

I wonder how to translate the above tf.train.Example into tfds.features? Thanks a lot!

**What I've tried so far**
I have searched the official doc for quite some time but cannot find the answer. There is a `Scalar` class in `tfds.features`, which I assume could be used to decode `Int64List`. But How can I decode the `BytesList`

**It would be nice if...**
It would be nice if the the official doc [here](https://www.tensorflow.org/datasets/external_tfrecord) gives more description when the user don't control the generation pipeline, which is often the case. Because if the user does control the generation pipeline, the user can simply turn to the custom datasets writing doc [here](https://www.tensorflow.org/datasets/add_dataset)

**Environment information**
* `tensorflow-datasets`/`tfds-nightly` version: 4.8.2
* `tensorflow`/`tensorflow-gpu`/`tf-nightly`/`tf-nightly-gpu` version: 2.11.0

**Comment by Conchylicultor** [comment]
In your case, you could have something like:

```python
features = tfds.features.FeaturesDict({
    'image': tfds.features.Image(),  # << Ideally best if you add the `shape=(h, w, c)` info
    'caption': tfds.features.Text(),
    'height': tf.int32,
    'width': tf.int32,
})
```

Then you can load the data either with [tfds.folder_dataset.write_metadata](https://www.tensorflow.org/datasets/api_docs/python/tfds/folder_dataset/write_metadata) or directly with:

```python
ds = tf.data.TFRecordDataset()
ds = ds.map(features.deserialize_example)
```

`batch`, `shuffle `,... will work as expected.

The TFDS metadata can be useful though if you want better split control (like `tfds.load(..., split='train[:75%]')`)

**Comment by zeyuwang615** [comment]
Thanks! it works! I thought I need to use some raw bytes reading class and then manually translate the encoded images and texts after deserializing examples, but `tfds.features.Image` and `tfds.features.Text` worked perfectly. 

Yeah, my biggest worry about the naive `tf.data.TFRecordDataset` is that without those metadata, it is impossible to split the dataset. Say, I have 8 processes for 8 devices, where each one is responsible for a 1/8 subsplit. It can be easily implemented by `process_splits = tfds.even_splits(split, process_count)` in TFDS, but not so for `tf.data.TFRecordDataset`.

**Comment by zeyuwang615** [comment]
I do have another problem regarding the split_info computation. As suggested by the doc, `compute_split_info.py` does the job, but I cannot find it. So I choose [tfds.folder_dataset.compute_split_info](https://www.tensorflow.org/datasets/api_docs/python/tfds/folder_dataset/compute_split_info), shown as follows.
```

filename_template = tfds.core.ShardedFileTemplate(
    data_dir='gs://path/to/data',
    template='{SHARD_INDEX}.{FILEFORMAT}',
    dataset_name=dataset_name,
    filetype_suffix='tfrecord',
)
split_infos = tfds.folder_dataset.compute_split_info(
     out_dir=out_dir,
     filename_template=filename_template,
)
```

But I cannot find any file in the out_dir. Moreover, the terminal output looks strange, which is repeated string of the following

> FilenameInfo(dataset_name='...', split=None, filetype_suffix='tfrecord', shard_index=00001, num_shards=None, filename_template=ShardedFileTemplate(data_dir=PosixGPath('...'), template='{SHARD_INDEX}.{FILEFORMAT}', dataset_name='...', split=None, filetype_suffix='tfrecord'))]

I wonder what is the right way to compute split_info?
